bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/DateExpression.java

```diff
@@ -0,0 +1,171 @@
+
+        testDateCal.add(Calendar.SECOND, -1);
+
+        Date timeAfter = getTimeAfter(testDateCal.getTime());
+
+        return ((timeAfter != null) && (timeAfter.equals(originalDate)));
+    }
+
+    public static boolean isValidExpression(String dateExpression) {
+        try {
+            new DateExpression(dateExpression);
+        } catch (ParseException pe) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public Date getFinalFireTime() {
+        if (candidates.isEmpty()) {
+            try {
+                parseExpression(expression);
+            } catch (ParseException e) {
+                logger.error("An exception occurred while parsing the expression : '{}'", e.getMessage());
+            }
+        }
+
+        if (candidates.isEmpty()) {
+            return null;
+        }
+
+        return candidates.get(candidates.size() - 1);
+    }
+
+    @Override
+    protected void validateExpression() throws IllegalArgumentException {
+        // Nothing to do here
+    }
+
+    @Override
+    protected void populateWithSeeds() {
+        // Nothing to do here
+    }
+
+    @Override
+    protected DateExpressionPart parseToken(String token, int position) throws ParseException {
+        return new DateExpressionPart(token);
+    }
+
+    protected class DateExpressionPart extends AbstractExpressionPart {
+
+        private Date theDate;
+
+        public DateExpressionPart(String s) throws ParseException {
+            super(s);
+        }
+
+        public Date getDate() {
+            return theDate;
+        }
+
+        @Override
+        public void parse() throws ParseException {
+            // Unfortunately, the time zone formats available to SimpleDateFormat (Java 6 and earlier) are not ISO 8601
+            // compliant. SimpleDateFormat understands time zone strings like "GMT+01:00" or "+0100", the latter
+            // according to RFC # 822.
+            //
+            // Even if Java 7 added support for time zone descriptors according to ISO 8601, SimpleDateFormat is still
+            // not able to properly parse a complete date string, as it has no support for optional parts.
+            //
+            // Reformatting the input string using regexp is certainly one possibility, but the replacement rules are
+            // not as simple
+            //
+            // Some time zones are not full hours off UTC, so the string does not necessarily end with ":00".
+            // ISO8601 allows only the number of hours to be included in the time zone, so "+01" is equivalent to
+            // "+01:00". ISO8601 allows the usage of "Z" to indicate UTC instead of "+00:00".
+            // The easier solution is to use the data type converter in JAXB, since JAXB must be able to parse
+            // ISO8601 date string according to the XML Schema specification.
+
+            try {
+                Calendar cal = DatatypeConverter.parseDateTime(expression);
```

Comment:
Just as a side-mark: The ```javax.xml.bind``` package is not available in the Compact 2 profile, which will make it yet harder to run any ESH based products on really weak hardware. So if there is another feasible way, I'd suggest to not draw in a dependency to jaxb here. Although I clearly understand how handy it is here...!